app/builders/v2/report_builder.rb

```diff
@@ -15,7 +15,13 @@ def initialize(account, params)
   end
 
   def timeseries
-    send(params[:metric])
+    metric_name = params[:metric]
+
+    if metric_name && respond_to?(metric_name)
+      send(metric_name)
+    else
+      Rails.logger.error "Invalid metric: #{metric_name}"
+    end
```

Comment:
I think you would need to return an empty object. Otherwise, it would additional errors in the build method.

@vishnu-narayanan Can you please add a spec for the same?

Reply:
@pranavrajs Thank you for the review. Let me add the spec.

Reply:
@pranavrajs Thank you for the review. Let me add the spec.